cmd/updater/subcommands/run/command.go

```diff
@@ -7,43 +7,76 @@
 package run
 
 import (
-	"fmt"
+	"context"
+
+	"go.uber.org/fx"
 
 	"github.com/DataDog/datadog-agent/cmd/updater/command"
-	"github.com/DataDog/datadog-agent/pkg/updater"
+	"github.com/DataDog/datadog-agent/comp/core"
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log/logimpl"
+	"github.com/DataDog/datadog-agent/comp/core/secrets"
+	"github.com/DataDog/datadog-agent/comp/core/sysprobeconfig/sysprobeconfigimpl"
+	"github.com/DataDog/datadog-agent/comp/remote-config/rcservice"
+	"github.com/DataDog/datadog-agent/comp/remote-config/rcservice/rcserviceimpl"
+	"github.com/DataDog/datadog-agent/comp/remote-config/rctelemetryreporter/rctelemetryreporterimpl"
+	"github.com/DataDog/datadog-agent/pkg/config/remote/service"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+
+	"github.com/DataDog/datadog-agent/comp/updater/localapi"
+	"github.com/DataDog/datadog-agent/comp/updater/localapi/localapiimpl"
+	"github.com/DataDog/datadog-agent/comp/updater/updater/updaterimpl"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
 
 	"github.com/spf13/cobra"
 )
 
+type cliParams struct {
+	command.GlobalParams
+}
+
 // Commands returns the run command
 func Commands(global *command.GlobalParams) []*cobra.Command {
 	runCmd := &cobra.Command{
 		Use:   "run",
 		Short: "Runs the updater",
 		Long:  ``,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return run(global.Package)
+			return runFxWrapper(&cliParams{
+				GlobalParams: *global,
+			})
 		},
 	}
 	return []*cobra.Command{runCmd}
 }
 
-func run(pkg string) error {
-	orgConfig, err := updater.NewOrgConfig()
-	if err != nil {
-		return fmt.Errorf("could not create org config: %w", err)
-	}
-	u, err := updater.NewUpdater(orgConfig, pkg)
-	if err != nil {
-		return fmt.Errorf("could not create updater: %w", err)
-	}
-
-	u.Start()
-	defer u.Stop()
+func runFxWrapper(params *cliParams) error {
+	ctx := context.Background()
+	return fxutil.OneShot(
+		run,
+		fx.Provide(func() context.Context { return ctx }),
+		fx.Supply(core.BundleParams{
+			ConfigParams:         config.NewAgentParams(params.GlobalParams.ConfFilePath),
+			SecretParams:         secrets.NewEnabledParams(),
+			SysprobeConfigParams: sysprobeconfigimpl.NewParams(),
+			LogParams:            logimpl.ForDaemon("UPDATER", "updater.log_file", pkgconfig.DefaultUpdaterLogFile),
```

Comment:
`pkgconfig.DefaultUpdaterLogFile)` could we replace this with the usage of the `config.Component` ?

Reply:
Do you have an example where this is done ? I'm not sure to follow (IIRC I copy/pasted existing patterns)